File: dotnet-back-end/ktdi-app/Model/RoomBooking.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Authorize]
[ApiController]
[Route("api/[controller]")]
public class RoomBooking
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int BookingId { get; set; }

    public int UserId { get; set; }

    public int RoomId { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public string ApprovalStatus { get; set; } = "Pending";

    public string PaymentDetails { get; set; }

    [ForeignKey("UserId")]
    public Authentication User { get; set; }

    [ForeignKey("RoomId")]
    public RoomRegistration Room { get; set; }
}

File: springboot-back-end/src/main/java/com/model/RoomBooking.java
package com.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "RoomBooking")
@Data
public class RoomBooking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "booking_id")
    private Long bookingId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private Authentication user;

    @ManyToOne
    @JoinColumn(name = "room_id", nullable = false)
    private RoomRegistration room;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "approval_status")
    private String approvalStatus = "Pending";

    @Column(name = "payment_details")
    private String paymentDetails;
}